# Identification of AD-specific *cis* regulatory elements regulated by ZEB1 and MAFB in neurons and microglia using single nuclei multiomics

![Figrue1](https://github.com/aanderson54/scMultiomics_AD/blob/main/images/Figure1-2.png)


Cell type-specific transcriptional differences between brain tissues from donors with Alzheimerâ€™s
Disease (AD) and unaffected controls have been well-documented, but few studies have rigorously
interrogated the regulatory mechanisms responsible for these alterations. We performed single cell
multiomics (RNA-seq+ATAC-seq) on 105,332 nuclei isolated from cortical tissues from 7 AD and 8
unaffected donors to identify candidate cis-regulatory elements (CREs) involved in AD-associated
transcriptional changes. We detected 319,861 significant correlations, or links, between gene
expression and cell-type specific transposase accessible regions that are enriched for active CREs.
Among these, 40,831 were unique to AD tissues. Validation experiments confirmed the activity of
many regions, including several candidate regulators of APP expression. We identified ZEB1 and
MAFB as transcription factors playing important roles in AD-specific gene regulation in neurons and
microglia, respectively. Links were globally enriched for heritability of AD risk, specifically in
microglia, and previously identified active regulatory regions.


## Analysis

To view a vignette of all analyses click [here](https://aanderson54.github.io/scMultiomics_AD/)

Additional pre-processing scripts can be found in the scripts directory.

